packages/tupaia-web/src/api/queries/useUser.ts

```diff
@@ -12,6 +12,6 @@ export const useUser = () => {
 
   return {
     ...userResponse,
-    isLoggedIn: data?.name !== undefined,
+    isLoggedIn: data?.email ? true : false, // when the name is public
```

Comment:
Oh yes the reason I changed this to name!== undefined is because in the new server it simply returns {} if the user is logged out. I think what you've done is ideal anyway since it will work with both the old and the new server.